qa/suites/rbd/nvmeof/workloads/nvmeof_initiator.yaml

```diff
@@ -18,10 +16,30 @@ tasks:
     no_coverage_and_limits: true
     clients:
       client.2:
-        - rbd/nvmeof_initiator.sh
-        - rbd/nvmeof_basic_tests.sh
-        - rbd/nvmeof_fio_test.sh
+        - rbd/nvmeof_setup_subsystem.sh
     env:
       RBD_POOL: mypool
-      RBD_IMAGE: myimage
+      RBD_IMAGE_PREFIX: myimage
+
+- parallel:
+    - rbd-iostat
+    - initiator-tests-and-fio
+
+rbd-iostat:
+- cephadm.shell:
+    host.a:
+      - rbd perf image iostat mypool --iterations 120 # (iterations ~= fio-runtime / 5)
```

Comment:
RBD iostat is for `nvmeof_fio_test.sh` only, not for `nvmeof_basic_tests.sh`. I noticed it doesn't start printing anything until it detects some I/O (which happens when fio starts) so I added it in parallel to both. 

@caroav asked me to add this iostat to fio, to ensure I/O can be seen happening in RBD image level, along with iostat we see for nvme drives.

Reply:
@idryomov we added the rbd level stats, because we see sometimes that fio reports io activity, while we know that it is not really doing writes. For example, if we blocklisted, and didn't recycle the cluster contexts, fio was still reporting ios. That is probably because we didn't use the direct flag in fio. In any case, we want to double check that ios really happens, and we don't trust only what fio reports.

Reply:
> RBD iostat is for `nvmeof_fio_test.sh` only, not for `nvmeof_basic_tests.sh`.

I think it belongs in `nvmeof_fio_test.sh` then, not in the common yaml.  The small delay that is experienced until enough I/O accumulates to be reported by the OSDs shouldn't matter for a 10-minute fio run.

Reply:
> That is probably because we didn't use the direct flag in fio.

Yes, that would do it ;)

> In any case, we want to double check that ios really happens, and we don't trust only what fio reports.

The job doesn't assert on any of these numbers -- it would be green even if no writes happen at the RBD level.  Do you expect each run to be reviewed manually (i.e. someone opening `teuthology.log` file)?

Reply:
> The job doesn't assert on any of these numbers -- it would be green even if no writes happen at the RBD level. Do you expect each run to be reviewed manually (i.e. someone opening teuthology.log file)?
No. But if we do open the log to see what happened, then I would like to be sure that I didn't get a false positive. In any case, I do think that at some point we need to make sure that we have IOs. I'm not sure how to do it (like check if > 0.0 is enough).

Reply:
If there's no initial I/O detected, then rbd's iostat would keep waiting and can be timed-out if no I/O shows-up on RBD level. That can be a basic verification for now?
`timeout $RUNTIME rbd perf image iostat mypool --iterations $iterations`

Or I can add an assertion of the number of images in the RBD's iostat output? And exit with an error if it's zero images in the output. 
```shell
wait_period=10
iterations=$(( $RUNTIME / $wait_period ))
for i in `seq 1 $iterations` ; do
  output=$(timeout 2m rbd perf image iostat mypool --format json | jq length)
  if [ "$output" = "0" ]; then
        echo "No I/O activity in any of the RBD images!"
        exit 1
  fi
  sleep $wait_period
done
```
Though json output this might effect the readability of rbd iostat when we need to read logs for debugging, especially when we have 128+ images.

Reply:
@VallariAg  what is the timeout value? If you want to use that, we just need to make sure to do it, in the places that we're sure that we should see ios.

Reply:
> Or I can add an assertion of the number of images in the RBD's iostat output? And exit with an error if it's zero images in the output.

It's up to you, but I have to admit that I have never seen a test that attempted to check whether I/O is going through with iostat.  The most sure of doing that is writing some known data, tearing down the device, setting it back up and reading the data back.  The data can be predefined or randomly generated at the beginning of the test.  Anything else just doesn't provide the same assurance -- iostat output, be at high-level or low-level, is just a bunch of numbers after all...

Reply:
> what is the timeout value? If you want to use that, we just need to make sure to do it, in the places that we're sure that we should see ios.

@caroav the timeout value is same as FIO runtime. The logic was that some I/O should appear during fio so if the iostat command is still waiting for it when fio is done then something is wrong and timeout will exit with an error. 


> It's up to you, but I have to admit that I have never seen a test that attempted to check whether I/O is going through with iostat. The most sure of doing that is writing some known data, tearing down the device, setting it back up and reading the data back. The data can be predefined or randomly generated at the beginning of the test. Anything else just doesn't provide the same assurance -- iostat output, be at high-level or low-level, is just a bunch of numbers after all...

I see, maybe asserting on iostat is not the right way to check for I/O then. Not sure if we should remove RBD iostat now or keep it without asserts?

Reply:
Keep it without assert IMHO, unless @idryomov has a strong objection to this?

Reply:
No objection from me.